### PR TITLE
Task/dat 1155 support data v1 migration

### PR DESCRIPTION
**Pull request by hb0**
Not much changed, I mostly moved a model to the `libs/model` class in `serialization` and upgraded to the new serialization version.

These changes are still relevant and have nothing to do with the "v1 data migration". They are relevant to have the pipelines be based on already deserialized data (not raw data).

**TODO:** => I'll change the name during Git Squash when merging, so it does not indicate that it has something to do with v1 support.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ subprojects {
     slf4jVersion = '1.7.29'
     commonsLangVersion = '3.12.0'
     logbackVersion = '1.2.5'
-    cyfaceSerializationVersion = '1.4.2'
+    cyfaceSerializationVersion = '1.5.0-beta1'
```
Comment (hb0) [inline build.gradle:85]: TODO

File: libs/api/src/main/java/de/cyface/api/MeasurementIterator.java [hunk 2]
```diff
@@ -157,7 +157,7 @@ public void handle(final JsonObject document) {
             // Handle bucket of next measurement (i.e. all buckets of previous measurement loaded)
             // System.out.println("MeasurementStream.handle(next measurements buckets) -> resolveNext");
             resolveNext(cachedBuckets, measurementId, bucket);
-        } catch (ParseException e) {
+        } catch (ParseException | RuntimeException e) {
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/MeasurementIterator.java:152]: Without this the provider API did not respond when it received a request to load data in an unsupported version as the Runtime Exception was not passed to the failure handler.

Not it responds with 500, which is okay.

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalImpl.java [hunk 3]
```diff
@@ -1,115 +0,0 @@
...
-import io.vertx.core.json.JsonArray;
-import io.vertx.core.json.JsonObject;
-
-/**
- * This class defines methods required by all {@link MeasurementRetrievalStrategy} implementations.
- *
- * @author Armin Schnabel
- * @since 6.0.0
- */
-public abstract class MeasurementRetrievalImpl implements MeasurementRetrievalStrategy {
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalImpl.java:40]: This class was never used and the method implementations are already `default` implementations in the interface.

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalStrategy.java [hunk 4]
```diff
@@ -68,8 +68,8 @@ public interface MeasurementRetrievalStrategy {
     default MetaData metaData(final JsonObject document) {
         final var metaData = document.getJsonObject("metaData");
         final var version = metaData.getString("version");
-        Validate.isTrue(version.equals(MetaData.CURRENT_VERSION),
-                "Encountered data in invalid format. Only Cyface Format Version 1.0.0 is supported!");
+        Validate.isTrue(version.matches(MetaData.SUPPORTED_VERSIONS),
+                "Encountered data in an unsupported deserialized format: %s", version);
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalStrategy.java:73]: All deserialized data with the same API version should be supported. Right now we only have 2.0.0 in V2.

File: libs/api/src/main/java/de/cyface/api/model/TrackBucket.java [hunk 5]
```diff
@@ -1,112 +0,0 @@
...
-import de.cyface.model.Track;
-import io.vertx.core.json.JsonObject;
-
-/**
- * The mongo database "bucket" which contains a slice of a track.
- *
- * @author Armin Schnabel
- * @since 6.0.0
- */
-public class TrackBucket {
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/model/TrackBucket.java:35]: This model moved to `serialization/libs/model`, see other PR.

File: libs/api/src/main/java/de/cyface/api/MeasurementIterator.java [hunk 6]
```diff
@@ -227,7 +219,7 @@ public Future<Measurement> next() {
      */
     private void resolveNext(final List<TrackBucket> buckets, final MeasurementIdentifier nextMeasurementId,
             final TrackBucket nextBucket) {
-        final var measurement = measurement(buckets);
+        final var measurement = new Measurement(buckets);
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/MeasurementIterator.java:222]: This constructor code moved to serialization/libs/model/Measurement, as you can see (also by the deleted lines below)

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithSensorData.java [hunk 7]
```diff
@@ -67,6 +72,8 @@ public TrackBucket trackBucket(final JsonObject document) throws ParseException
         final var directions = new ArrayList<>(point3D(directionsDocuments));
 
         final var track = new Track(locationRecords, accelerations, rotations, directions);
-        return new TrackBucket(trackId, bucket, track, metaData);
+        // noinspection SpellCheckingInspection
+        final var date = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").parse(bucket.getString("$date"));
+        return new TrackBucket(trackId, date, track, metaData);
```
Comment (hb0) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithSensorData.java:77]: I have to unpack the `JsonObject` to `Date` now here, as the `serialization/libs/model` has no dependency to `Vertx.JsonObject`.

**Review by hb0 — COMMENTED** [review]
pre-reviewed

File: build.gradle [hunk 8]
```diff
@@ -82,7 +82,7 @@ subprojects {
     slf4jVersion = '1.7.29'
     commonsLangVersion = '3.12.0'
     logbackVersion = '1.2.5'
-    cyfaceSerializationVersion = '1.4.2'
+    cyfaceSerializationVersion = '1.5.0-beta2'
```
Comment (hb0) [inline build.gradle:85]: TODO

File: libs/api/src/test/java/de/cyface/api/MeasurementIteratorTest.java [hunk 9]
```diff
@@ -1,282 +0,0 @@
...
-import io.vertx.core.streams.ReadStream;
-
-/**
- * Tests the inner workings of {@link MeasurementIterator}.
- *
- * @author Armin Schnabel
- * @version 6.0.0
- * @since 1.0.0
- */
-public class MeasurementIteratorTest {
```
Comment (hb0) [inline libs/api/src/test/java/de/cyface/api/MeasurementIteratorTest.java:58]: moved to `serialization.libs.model.MeasurementTest`

File: build.gradle [hunk 10]
```diff
@@ -82,7 +82,7 @@ subprojects {
     slf4jVersion = '1.7.29'
     commonsLangVersion = '3.12.0'
     logbackVersion = '1.2.5'
-    cyfaceSerializationVersion = '2.0.0'
+    cyfaceSerializationVersion = '2.1.0-beta1'
```
Comment (hb0) [inline build.gradle:85]: TODO

File: libs/api/src/main/java/de/cyface/api/model/TrackBucket.java (hunk 5 above)
Comment (hb0) [reply libs/api/src/main/java/de/cyface/api/model/TrackBucket.java:35]: `serialization` 2.1.0 now contains this model, the Upgrade to 2.1.0 was done in the previous commit (due to rebasing this moved)

**Review by muthenberg — APPROVED** [review]

**Pull request merged**